### PR TITLE
Adds support for RedHat OpenShift clusters

### PR DESCRIPTION
**Pull request by etichy**
Our collector should be able to run on openshift cluster. 

Limitations: 
- ebpf is not running correctly - https://github.com/open-telemetry/opentelemetry-network/issues/254

### CODE REVIEW

File: deploy/helm/templates/node-collector-daemon-set.yaml [hunk 1]
```diff
@@ -31,11 +31,20 @@ spec:
     spec:
       terminationGracePeriodSeconds: {{ .Values.otel.node_collector.terminationGracePeriodSeconds }}
       serviceAccountName: {{ include "common.fullname" . }}
+      hostNetwork: true
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/templates/node-collector-daemon-set.yaml:34]: Do we need to be on host metric everywhere or only on openshift?

File: deploy/helm/templates/openshift/openshift-role-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.openshift.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "common.fullname" (tuple . "-openshift-role-binding") }}
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/templates/openshift/openshift-role-binding.yaml:5]: Shoud we name that this RB is for kube-state-metrics?

Comment (etichy) [reply deploy/helm/templates/openshift/openshift-role-binding.yaml:5]: it has collector name in prefix, right?
image

File: deploy/helm/templates/node-collector-daemon-set.yaml (hunk 1 above)
Comment (etichy) [reply deploy/helm/templates/node-collector-daemon-set.yaml:34]: this is due to INFRA nodes, we can make it conditional only for openshift.

Comment (gantrior) [reply deploy/helm/templates/node-collector-daemon-set.yaml:34]: Why do we need to mount node collector to host network interface? Even in openShift?

Comment (etichy) [reply deploy/helm/templates/node-collector-daemon-set.yaml:35]: hostNetwork option solves issues with access to journal logs and wr access to checkpoints folder. We decided that as first draft we go out with this not optimal solution and in operator we will deal with it properly.

**Review by gantrior — APPROVED** [review]

**Pull request merged**